### ISSUE TITLE
Make cache policy support variables

### ISSUE THREAD

**Issue by ilim-ablyaev** [firecow/gitlab-ci-local#1091]
In GitLab CI/CD, version 16.5.1 introduced the capability to control a job's cache policy using variables (https://docs.gitlab.com/ee/ci/caching/#use-a-variable-to-control-a-jobs-cache-policy). However, this functionality is missing in gitlab-ci-local.

**Comment by amenella** [comment]
Hi, just ran into this problem too (nice coincidence :sweat_smile: )

The actual behavior by gitlab-ci-local is to refuse to run the job with following error message:
```bash
my_job_name cache[0].policy is not 'pull', 'push' or 'pull-push'
```

however the varibale is properly defined on the job:
```yaml
my_job_name:
  stage: some_stage
  variables:
    CACHE_POLICY: pull-push
  cache:
    - key: my_cache_key
      paths:
        - some/path/to/cache
      policy: $CACHE_POLICY
```

hope this helps

PS: thanks a lot for gitlab-ci-local, this is a really helpful tool :pray:

**Comment by ilim-ablyaev** [comment]
Hi, @amenella 
I've made PR with fix) let's hope that it will be merged soon!) #1092
